src/pkg/api/composition.go

```diff
@@ -0,0 +1,96 @@
+package api
+
+import (
+	"encoding/json"
+	"hms/gateway/pkg/config"
+	"hms/gateway/pkg/docs/service/composition"
+	"hms/gateway/pkg/docs/types"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+
+	"hms/gateway/pkg/docs/model"
+	"hms/gateway/pkg/docs/service"
+	"hms/gateway/pkg/errors"
+)
+
+type CompositionHandler struct {
+	service *composition.CompositionService
+}
+
+func NewCompositionHandler(docService *service.DefaultDocumentService, cfg *config.Config) *CompositionHandler {
+	return &CompositionHandler{
+		service: composition.NewCompositionService(docService, cfg),
+	}
+}
+
+func (h CompositionHandler) Create(c *gin.Context) {
+	ehrId := c.Param("ehrid")
+	if h.service.Doc.ValidateId(ehrId, types.EHR) == false {
+		c.AbortWithStatus(http.StatusNotFound)
+		return
+	}
+
+	data, err := ioutil.ReadAll(c.Request.Body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": "Request body error"})
+		return
+	}
+
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			log.Panic("Cant close body request")
+		}
+	}(c.Request.Body)
+
+	var request model.CompositionCreateRequest
```

Comment:
Here we can use model.Composition. CompositionCreateRequest is not required

Reply:
+